### ISSUE TITLE
NoMethodError: undefined method `root=' for nil:NilClass

### ISSUE THREAD

**Issue by mrhistory** [jruby/warbler#366]
Hi,

I am trying to use Warbler 2.0.0.rc2 under JRuby 2.2.3. I get the following error:
```
sovsys:dekt jchambers$ jruby -S warble war --trace
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/rubyzip-1.1.7/lib/zip.rb:35: warning: ObjectSpace impacts performance. See http://wiki.jruby.org/PerformanceTuning#dont-enable-objectspace
warble aborted!
NoMethodError: undefined method `root=' for nil:NilClass
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/traits/war.rb:53:in `default_webxml_config'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/traits/war.rb:27:in `before_configure'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/traits.rb:29:in `block in before_configure'
org/jruby/RubyArray.java:1567:in `each'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/traits.rb:29:in `before_configure'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/config.rb:211:in `initialize'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/task.rb:48:in `initialize'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/application.rb:27:in `load_rakefile'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/rake-10.5.0/lib/rake/application.rb:77:in `block in run'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/rake-10.5.0/lib/rake/application.rb:176:in `standard_exception_handling'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/rake-10.5.0/lib/rake/application.rb:75:in `run'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/lib/warbler/application.rb:77:in `run'
/Users/jchambers/.rvm/gems/jruby-2.2.3/gems/warbler-2.0.0.rc2/bin/warble:11:in `<top>'
org/jruby/RubyKernel.java:958:in `load'
/Users/jchambers/.rvm/gems/jruby-2.2.3/bin/warble:23:in `<top>'
```
Gem Environment:
sovsys:dekt jchambers$ gem env
RubyGems Environment:
  - RUBYGEMS VERSION: 2.5.1
  - RUBY VERSION: 2.3.0 (2016-02-29 patchlevel 0) [java]
  - INSTALLATION DIRECTORY: /Users/jchambers/.rvm/gems/jruby-2.2.3
  - USER INSTALLATION DIRECTORY: /Users/jchambers/.gem/jruby/2.3.0
  - RUBY EXECUTABLE: /Users/jchambers/.rvm/rubies/jruby-2.2.3/bin/jruby
  - EXECUTABLE DIRECTORY: /Users/jchambers/.rvm/gems/jruby-2.2.3/bin
  - SPEC CACHE DIRECTORY: /Users/jchambers/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Users/jchambers/.rvm/rubies/jruby-2.2.3/etc
  - RUBYGEMS PLATFORMS:
    - ruby
    - universal-java-1.8
  - GEM PATHS:
     - /Users/jchambers/.rvm/gems/jruby-2.2.3
     - /Users/jchambers/.rvm/gems/jruby-2.2.3@global
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
     - "install" => "--no-rdoc --no-ri --env-shebang"
     - "update" => "--no-rdoc --no-ri --env-shebang"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /Users/jchambers/.rvm/gems/jruby-2.2.3/bin
     - /Users/jchambers/.rvm/gems/jruby-2.2.3@global/bin
     - /Users/jchambers/.rvm/rubies/jruby-2.2.3/bin
     - /Users/jchambers/.rvm/bin
     - /usr/local/bin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin

Thanks,
Jim

**Comment by erkmos** [comment]
I ran into this issue as well. The issue is that the implementation for method_missing in OpenStruct changed in MRI 2.3.x, it used to create a new instance when calling a non-existing method with an argument. But it no longer does that (which breaks call-chaining when not directly assigning values).

Change is here: https://github.com/jruby/jruby/commit/d6f4cc1539b4c1b2a6df4b6e0347a1c16a4d3989#diff-743606ad57ba17094bf4fbc504975af0L185

I fixed this by just overloading it with the old implementation. Might submit a pull request later if I can get the test suite working on OSX, not much luck so far. :(
